internal/provider/models/processors/test/metrics_tag_cardinality_limit_test.go

```diff
@@ -0,0 +1,186 @@
+					}`,
+				ExpectError: regexp.MustCompile("Attribute tags list must contain at most 10 elements"),
+			},
+
+			// Error: length validation for exclude_tags
+			{
+				Config: GetCachedConfig(cacheKey) + `
+					resource "mezmo_metrics_tag_cardinality_limit_processor" "my_processor" {
+						pipeline_id = mezmo_pipeline.test_parent.id
+						value_limit = 50
+						exclude_tags = [""]
+					}`,
+				ExpectError: regexp.MustCompile("Attribute exclude_tags.* string length must be at least 1"),
+			},
+			{
+				Config: GetCachedConfig(cacheKey) + `
+					resource "mezmo_metrics_tag_cardinality_limit_processor" "my_processor" {
+						pipeline_id = mezmo_pipeline.test_parent.id
+						value_limit = 50
+						exclude_tags = ["` + strings.Repeat("x", 101) + `"]
+					}`,
+				ExpectError: regexp.MustCompile("Attribute exclude_tags.* string length must be at most 100"),
+			},
+			{
+				Config: GetCachedConfig(cacheKey) + `
+					resource "mezmo_metrics_tag_cardinality_limit_processor" "my_processor" {
+						pipeline_id = mezmo_pipeline.test_parent.id
+						value_limit = 50
+						exclude_tags = ["x", "x", "x", "x", "x", "x", "x", "x", "x", "x", "x"]
+					}`,
+				ExpectError: regexp.MustCompile("Attribute exclude_tags list must contain at most 10 elements"),
+			},
+
+			// Error: action is an enum
+			{
+				Config: GetCachedConfig(cacheKey) + `
+					resource "mezmo_metrics_tag_cardinality_limit_processor" "my_processor" {
+						pipeline_id = mezmo_pipeline.test_parent.id
+						value_limit = 50
+						action = "nope"
+					}`,
+				ExpectError: regexp.MustCompile(`Attribute action value must be one of: \["drop_tag" "drop_event"\]`),
+			},
+
+			// Error: mode is an enum
+			{
+				Config: GetCachedConfig(cacheKey) + `
+					resource "mezmo_metrics_tag_cardinality_limit_processor" "my_processor" {
+						pipeline_id = mezmo_pipeline.test_parent.id
+						value_limit = 50
+						mode = "nope"
+					}`,
+				ExpectError: regexp.MustCompile(`Attribute mode value must be one of: \["exact" "probabilistic"\]`),
+			},
+
+			// Create with defaults
+			{
+				Config: GetCachedConfig(cacheKey) + `
+					resource "mezmo_metrics_tag_cardinality_limit_processor" "my_processor" {
+						title = "title"
+						description = "desc"
+						pipeline_id = mezmo_pipeline.test_parent.id
+						value_limit = 50
+					}`,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestMatchResourceAttr(
+						"mezmo_metrics_tag_cardinality_limit_processor.my_processor", "id", regexp.MustCompile(`[\w-]{36}`)),
+
+					StateHasExpectedValues("mezmo_metrics_tag_cardinality_limit_processor.my_processor", map[string]any{
+						"pipeline_id":   "#mezmo_pipeline.test_parent.id",
+						"title":         "title",
+						"description":   "desc",
+						"generation_id": "0",
+						"inputs.#":      "0",
+						"mode":          "exact",
+						"action":        "drop_event",
+						"value_limit":   "50",
+						"%":             "11",
```

Comment:
What does "%" here refer to?

Reply:
it's the total number of key/vals.  Adding this to the assertion makes sure the server hasn't added properties we don't have covered.